src/xds/client.rs

```diff
@@ -359,13 +373,19 @@ impl AdsClient {
                     .join("; "),
             ),
         };
+
         info!(
             type_url = type_url.clone(),
             none = response.nonce,
             "sending {}",
             if error_detail.is_some() {
                 "NACK"
             } else {
+                if let Some(tx) = initial_xds_recvd {
+                    if let Err(err) = tx.send(()) {
```

Comment:
Is it safe to send multi times?

Reply:
We never get to that point; `send()` won't even be invoked when `tx` is `None`, and it gets set to None on L297 when `take` is called

Reply:
good to know that

Reply:
I'll add a comment explaining the mechanism; it's not super clear looking at it later